projects/bootloader/test/test_query.c

```diff
@@ -0,0 +1,280 @@
+                               &s_test_datagram_settings);
+  dispatcher_init(s_board_id);
+}
+
+void teardown_test(void) {}
+
+void test_query(void) {
+  // test max sized response (almost maxed, s_board_id is not the longest possible)
+  BootloaderConfig config = {
+    .controller_board_id = s_board_id,
+    .controller_board_name = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz12345678910",
+    .project_present = true,
+    .project_name = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz12345678910",
+    .project_info = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz12345678910",
+    .git_version = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz12345678910",
+  };
+  query_init(&config);
+
+  // "id" : [255]
+  // "name" : ["abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz12345678910"]
+  // "current_project" : ["abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz12345678910"]
+  // "project_info" : ["abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz12345678910"]
+  // "git_version" : ["abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz12345678910"]
+  size_t len = 262;
+  // exact query, so this is also the expected response
+  uint8_t data[262] = {
+    0x08, 0x02, 0x12, 0x3F, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B, 0x6C,
+    0x6D, 0x6E, 0x6F, 0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x61, 0x62,
+    0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71, 0x72,
+    0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38,
+    0x39, 0x31, 0x30, 0x1A, 0x3F, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B,
+    0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x61,
+    0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71,
+    0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+    0x38, 0x39, 0x31, 0x30, 0x22, 0x3F, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A,
+    0x6B, 0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A,
+    0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D, 0x6E, 0x6F, 0x70,
+    0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36,
+    0x37, 0x38, 0x39, 0x31, 0x30, 0x2A, 0x3F, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69,
+    0x6A, 0x6B, 0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79,
+    0x7A, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D, 0x6E, 0x6F,
+    0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x31, 0x32, 0x33, 0x34, 0x35,
+    0x36, 0x37, 0x38, 0x39, 0x31, 0x30,
+  };
+  // response
+  size_t expected_len = len;
+  uint8_t *expected_response = data;
```

Comment:
You set up all these variables and didn't do anything with them, did you forget to finish this test?